### PR TITLE
Speed up equalize transform: use bincount instead of histc

### PR DESCRIPTION
**Pull request by NicolasHug**
`bincount` is faster than `histc` on CPU, so this should speed up (~2X) the `equalize` transforms.

This is sort of a follow-up to https://github.com/pytorch/vision/pull/3334 and https://github.com/pytorch/vision/issues/3173

On GPU:

```py
In [16]: img = torch.randint(0, 256, size=(460 * 680,)).to('cuda')

In [17]: %timeit torch.bincount(img, minlength=256)
1.52 ms ± 25.3 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [18]: %timeit torch.histc(img.to(torch.float32), bins=256, min=0, max=255)
127 µs ± 1.07 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

on CPU (different machine):

```py
img = torch.randint(0, 256, size=(460 * 680,))

%timeit torch.histc(img.to(torch.float32), bins=256, min=0, max=255)
696 µs ± 10.1 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

%timeit torch.bincount(img, minlength=256)
289 µs ± 1.06 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

```

### CODE REVIEW

**Comment by fmassa** [comment]
There is a huge difference on CUDA. I think it would be worth opening an issue in PyTorch to point out that the implementation of `bincount` is suboptimal?

Also, it would be good to check the number of CUDA again, but this time using `torch.cuda.synchronize()` to get more accurate timings

File: torchvision/transforms/functional_tensor.py [hunk 1]
```diff
@@ -885,7 +885,10 @@ def autocontrast(img: Tensor) -> Tensor:
 
 
 def _scale_channel(img_chan):
-    hist = torch.histc(img_chan.to(torch.float32), bins=256, min=0, max=255)
+    if img_chan.is_cuda:
+        hist = torch.histc(img_chan.to(torch.float32), bins=256, min=0, max=255)
+    else:
+        hist = torch.bincount(img_chan.view(-1), minlength=256)
```
Comment (datumbox) [inline torchvision/transforms/functional_tensor.py:891]: Do you need to clip values between 0 and 255 to keep things consistent between cuda/cpu?

Comment (NicolasHug) [reply torchvision/transforms/functional_tensor.py:891]: Passing `minlength=256` ensures that `bincount` will count the number of values for all values in `[0, 255]`, so `histc` and `bincount` would be equivalent.

In the `histc` call we have `bins=256, min=0, max=255`, which IIUC assumes that the image is within `[0, 255]` already, is this not the case?

**Comment by NicolasHug** [comment]
I created a new "Improvement" label, in accordance with https://github.com/pytorch/vision/issues/3351#issuecomment-791482582

**Review by fmassa — APPROVED** [review]
Thanks!

**Pull request merged**